packages/core/src/execution-engine/node-execution-context/webhook-context.ts

```diff
@@ -172,4 +174,16 @@ export class WebhookContext extends NodeExecutionContext implements IWebhookFunc
 			itemIndex,
 		);
 	}
+
+	validateExecutionWaitingToken() {
+		const req = this.getRequestObject();
+		const token = req.query[WAITING_TOKEN_QUERY_PARAM];
+
+		if (typeof token !== 'string') return false;
+
+		const expectedToken = this.getExecutionWaitingToken();
+
+		const valid = crypto.timingSafeEqual(Buffer.from(token), Buffer.from(expectedToken));
```

Comment:
Can we make this backwards compatible so that executions created before the introduction of the HMAC mechanism can still be resumed?

Reply:
I think it could be doable by storing the current largest exec id (eg using a migration) and require the signature only for executions that are newer than the oldest one before this change. Requires some effort but prolly doable

Reply:
or we persist the expected signature as part of execution data and compare on resume…

Reply:
maybe also by using timestamps of execution

question is do we want have code for that included in n8n and affect every single execution farther
breaking will be only single execution of workflow started before instance upgrade, so probably not a lot